src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorShape.cs

```diff
@@ -447,15 +446,15 @@ public nint AdjustToNextIndex(in TensorShape destinationShape, nint linearOffset
                 nint length = lengths[rankIndex];
                 nint stride = strides[rankIndex];
 
-                nint index = ++indexes[rankIndex];
+                nint index = ++indexes[destinationShape.Lengths.Length - (i + 1)];
```

Comment:
`destinationShape.Lengths.Length - (i + 1)` should be cached as some `int destinationRankIndex` up next to `rankIndex`, just for clarity

Reply:
Fixed.